File: Journal System.py
# Name: Rusho Binnabi
# Date: 4/10/2022
# Project: Journal System
# Code File Updated: 4/16/2022 at 10:31 PM
# Contact Information: RushoBinnabi23@yahoo.com or 518-364-7101

import sys # imports the sys module for use in this program.
import os # imports the os module for use in this program.
import glob # import the glob module for use in this program.
import time # imports the time module for use in this program.

# this program acts as a journal system where it can create, read, list, and delete journal entries. 

beginTime = time.time() # creates a variable called beginTime which will have the beginning of the execution time in seconds.

def writeJournals(name): # this writeJournals() function takes a variable  called name as the argument which is the name of the journal entry files as a string.
                         # this function creates and writes new journal entries using the argument name as the file name of the journal entry as a string.
    characters = 35 # creates a variable called characters which is initialized to 35 and will be used as the number of characters that each line of the journal entry files will consist of after splitting.
    try: # the try block will try to run the code.
        file = open(name, "a") # creates a variable called file and by using the open() function, it will open the file specified in name in appending mode so the contents of the file can also be edited and the file can be created if it doesn't exist.
        journalData = input("\nEnter text: ") # prompts the user for text which will be entered into the journal entry file that was created using file and saves that prompt in the journalData variable that was created.
        while True: # the code in the while loop will run as long as the current string value isn't equal to "done" which means the user is done entering text into the file.
            journalData = [(journalData[i:i+characters]) for i in range(0, len(journalData), characters)] # the string values inside journalData will be split into the number of characters specified inside characters.
            file.writelines(journalData) # using the writelines() function, it writes the data inside journalData using file and journalData as the argument for the writelines() function.
            journalData = input() # journalData reads in the next input value.
            if journalData == "done": # if the current input value in journalData is equal to "done", then the code inside the if statement will be run.
                break # breaks out of the while loop.
        file.close() # using the close() function, it closes file.
        print("\nJournal entry created.") # tells the user that the journal entry was created.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
    except FileNotFoundError: # the code inside the except statement will run if the file that was specified in name doesn't exist.
        print("\nError. That journal entry doesn't exist.") # tells the user that the journal entry doesn't exist.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.

def readJournals(name): # this readJournals() function takes a variable called name as the argument which is the name of the journal entry file as a string.
                        # this function reads the contents of the journal entries using the argument name as the file name of the journal entry as a string.
    try: # the try block will try to run the code.
        file = open(name, "r") # creates a variable called file and by using the open() function, it will open the file specified in name in read mode so the contents of the file can be read.
        journalData = file.read() # creates a variable called journalData which will read the contents of file using the read() function.
        fileName = file.name # creates a variable called fileName which will have the name of the file inside file.
        file.close() # using the close() function, it closes file.
        print("\n" + "Journal entry: " + fileName + " contents: \n\n" + journalData) # prints the name of the file inside file and the data inside journalData.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
    except FileNotFoundError: # the code inside the except statement will run if the file that was specified in name doesn't exist.
        print("\nError. That journal entry doesn't exist.") # tells the user that the journal entry doesn't exist.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.

def exitProgram(): # this exitProgram() function exits the program.
    sys.exit(0) # using the exit() function from the sys module and the integer value of 0 which means the program stopped succesfully.

def listing(): # this listing() function lists the number of journal entry files and where the journal system program are saved.
    cwd = os.path.dirname(os.path.realpath(__file__)) # creates a variable called cwd which gets the full path of the current working directory and saves that path inside cwd as a string.
    print("\nThe Journal System is saved here: \n\n" + cwd + "\n") # tells the user where the journal system program is saved at.
    print("\nSaved Journal Entries:\n") # tells the user how many number of journal entries are saved.
    for file in glob.glob("*.txt"): # creates a for loop for every file that ends with the .txt extension using the glob() function from the glob module.
        print(file) # prints the name of the current file.
    time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
    os.system('cls') # using the system() function of the os module, it clears the screen.
    
def deleteJournals(name): # this deleteJournals() function takes a variable called name as the argument which is the name of the journal entry file as a string.
                          # this function deletes the journal entry file using the argument name as the file name of the journal entry as a string.
    file = open(name, "r") # creates a variable called file and by using the open() function, it will open the file specified in name in read mode so the name of the file can be read.
    fileName = file.name # creates a variable called fileName which will have the name of the file inside file.
    file.close() # using the close() function, it closes file.
    os.remove(name) # using the rmeove() function of the os module and name as it's argument, it removes the file that was specified in name.
    print("\n" + "Journal entry: " + fileName + " deleted.\n") # tells the user that the journal entry file inside file was deleted.
    time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
    os.system('cls') # using the system() function of the os module, it clears the screen.

userInput = input("\nJournal System\n\n1: Create Journal Entry\n2: Read Journal Entry\n3: Exit\n4: List Journal Entries and Location\n5: Remove Journal Entry\n\nEnter Choice: ") # prompts the user to enter a number which corresponds to a command and saves that number inside the userInput variable that was created.

while userInput != 3: # the code in the while loop will run as long as the user didn't enter a "3" that's stored inside userInput which means the user is done using the program.
    if userInput == "1": # if the value inside userInput is equal to 1, then the code inside the if statement will be run.
        name = input("\nEnter journal entry name: ") # prompts the user to enter a name for the journal entry file that will be created and saves that input in the name variable that was created.
        writeJournals(name) # calls the writeJournals() function and uses name as it's argument.
        endTime1 = time.time() # creates a variable called endTime1 which will have the end of the execution time in seconds.
        elaspedTime1 = endTime1 - beginTime # creates a variable called elaspedTime1 which will have the value of endTime1 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime1))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime1 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.

    elif userInput == "2": # or if the value inside userInput is equal to 2, then the code inside the elif statement will be run.
        name = input("\nEnter journal entry name: ") # prompts the user to enter a name for the journal entry file whose contents will be read and saves that input in the name variale that was created.
        readJournals(name) # calls the readJournals() function and uses name as it's argument.
        endTime2 = time.time() # creates a variable called endTime2 which will have the end of the execution time in seconds.
        elaspedTime2 = endTime2 - beginTime # creates a variable called elaspedTime2 which will have the value of endTime2 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime2))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime2 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.

    elif userInput == "3": # or if the value inside userInput is equal to 3, then the code inside the elif statement will be run.
        endTime3 = time.time() # creates a variable called endTime3 which will have the end of the execution time in seconds.
        elaspedTime3 = endTime3 - beginTime # creates a variable called elaspedTime3 which will have the value of endTime3 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime3))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime3 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
        exitProgram() # calls the exitProgram() function.

    elif userInput == "4": # or if the value inside userInput is equal to 4, then the code inside the elif statement will be run.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
        listing() # calls the listing() function.
        endTime4 = time.time() # creates a variable called endTime4 which will have the end of the execution time in seconds.
        elaspedTime4 = endTime4 - beginTime # creates a variable called elaspedTime4 which will have the value of endTime4 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime4))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime4 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
    
    elif userInput == "5": # or if the value inside userInput is equal to 5, then the code inside the elif statement will be run.
        name = input("\nEnter journal entry name: ") # prompts the user to enter a name for the journal entry file which will be deleted and saves that input in the name variale that was created.
        deleteJournals(name) # calls the deleteJournals() function and uses name as it's argument.
        endTime5 = time.time() # creates a variable called endTime5 which will have the end of the execution time in seconds.
        elaspedTime5 = endTime5 - beginTime # creates a variable called elaspedTime5 which will have the value of endTime5 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime5))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime5 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
    
    time.sleep(2) # slows down the execution speed of the program by 2 seconds using the sleep() function of the time module and the integer value of 2 as it's argument.

    choice = input("Do you want to use the Journal System again (y/n)? ") # prompts the user to see if they want to use the journal system program again, and saves that input in the choice variable that was created.
    time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
    os.system('cls') # using the system() function of the os module, it clears the screen.
    choice = choice.lower() # the values inside choice will be in lowercase.

    if choice == "y": # if the value inside choice was a "y" which means the user wants to use the program again, then the code inside the if statement will be run.
        userInput = input("\nJournal System\n\n1: Create Journal Entry\n2: Read Journal Entry\n3: Exit\n4: List Journal Entries and Location\n5: Remove Journal Entry\n\nEnter Choice: ") # prompts the user to enter a number again which corresponds to a command and saves that number inside the userInput variable that was created.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen. 
    elif choice == "n": # or if the value inside choice was a "n" which means the user doesn't want to use the program again, then the code inside the elif statement will be run.
        endTime6 = time.time() # creates a variable called endTime6 which will have the end of the execution time in seconds.
        elaspedTime6 = endTime6 - beginTime # creates a variable called elaspedTime6 which will have the value of endTime6 - beginTime which is the total amount of time it took for the program to run in seconds.
        print("\nExecution Time: ", time.strftime("%H:%M:%S", time.gmtime(elaspedTime6))) # shows how many seconds it took for the program to run using the gmtime() function of the time module and elaspedTime6 as it's argument.
        time.sleep(2) # slows down the execution speed of the program by 1 second using the sleep() function of the time module and the integer value of 1 as it's argument.
        os.system('cls') # using the system() function of the os module, it clears the screen.
        quit()

File: JournalSystem.cs
/*
* Name: Rusho Binnabi
* Date: 7/27/2022
* Project: Journal System
* Code File Updated: 7/27/2022 at 6:02 PM
* Contact Information: RushoBinnabi123@yahoo.com or 518-364-7101
*/

using System; // the System library gets used in this program.
using System.IO; // the System IO library gets used in this program.
using System.Threading; // the System Threading library gets used in this program.

namespace Journal { /* this Journal namespace contains all the methods, classes, etc needed for the program to run. */

    class JournalSystem { /* this JournalSystem class contains all the methods needed for this program to run. */

    // this program creates, deletes, reads, lists, etc journal entries based on what the user wants.

        public static string fileName; // a public static string variable called fileName gets created and will have the name of the journal entry file.
        public static int choice; // a public static integer variable called choice gets created and will have the integer number that signifies what function of the program that the user wants to use. 
        public static string content; // a public static string variable called content gets created and will have the contents of the journal entry files that will be read.
        public static string exitCondition; // a public static string variable called exitCondition gets created and will have the string value of y or n based on whether the user wants to use the program again or not.

        /*
        * this public static void resetValues() method resets all the values of the variables.
        * this method is void because it doesn't return anything.
        */

        public static void resetValues() {
            fileName = ""; // the value of fileName gets reset to an empty string.
            choice = 0; // the value of choice gets reset to 0.
            content = ""; // the value of content gets reset to an empty string.
            exitCondition = ""; // the value of exitCondition gets reset to an empty string.
        } // the end of the resetValues() method.

        /*
        * this public static void readJournalEntry() method reads a journal entry that the user wants to have read.
        * this method is void because it doesn't return anything.
        */

        public static void readJournalEntry() {
            Console.Write("Enter the name of the journal entry: "); // prompts the user to enter the name of the journal entry that they want to have read.
            fileName = Console.ReadLine(); // fileName gets the user input saved as a string using the Console.ReadLine() method.
            Console.WriteLine(""); // creates some line spacing.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.WriteLine("Contents of " + fileName + ":"); // shows the user what journal entry file is being read using the name from fileName.
            Console.WriteLine(""); // creates some line spacing.
            content = File.ReadAllText(fileName); // content gets the text read from the file in fileName using the File.ReadAllText() method.
            Console.WriteLine(content); // shows the user what the contents of the journal entry file was from content.
            Console.WriteLine(""); // creates some line spacing.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.WriteLine("Contents Read. Returning to Main Menu..."); // tells the user that the contents of the journal entry file has been read and the program is returning to the main menu.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.Clear(); // clears the window by calling the Console.Clear() method.
        } // the end of the readJournalEntry() method.

        /*
        * the public static void exitSystem() method exits the program.
        * this method is void because it doesn't return anything.
        */

        public static void exitSystem() {
            resetValues(); // calls the resetValues() method so the memory for the variables used in this program gets cleared.
            System.Environment.Exit(0); // exits the program with a value of 0 which means the program exited successfully.
        } // the end of the exitSystem() method.
        
        /*
        * this public static void createJournalEntry() method creates/writes a journal entry file.
        * this method is void because it doesn't return anything.
        */

        public static void createJournalEntry() {
            Console.Write("Enter the name of the journal entry: "); // prompts the user to enter the name of the journal entry file.
            fileName = Console.ReadLine(); // fileName gets the user input saved as a string using the Console.ReadLine() method.
            Console.WriteLine(""); // creates some line spacing.
            Console.Write("Enter the contents of the journal entry: "); // prompts the user to enter what the journal entry file will be containing.
            content = Console.ReadLine(); // content gets the text read from the user input as a string using the Console.ReadLine() method.
            File.WriteAllText(fileName, content); // creates/writes the journal entry file using the name of the file that the user entered from fileName and what the user wants to have written from content using the File.WriteAllText() method.
            Console.WriteLine(""); // creates some line spacing.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.WriteLine("Journal Entry created. Returning to Main Menu..."); // tells the user that the contents of the journal entry file has been written/created and the program is returning to the main menu.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.Clear(); // clears the window by calling the Console.Clear() method.
        } // the end of the createJournalEntry() method.

        /*
        * this public static void listJournalEntry() method lists all the journal entry files and shows the path to where the main system program is saved.
        * this method is void because it doesn't return anything.
        */

        public static void listJournalEntry() {
            DirectoryInfo di = new DirectoryInfo(Directory.GetCurrentDirectory()); // creates a new DirectoryInfo object called di that gets the path of the current directory using the Directory.GetCurrentDirectory() method as the argument for the DirectoryInfo object.
            FileInfo[] files = di.GetFiles("*.txt"); // creates a new FileInfo array object called files that gets all the files that have the ".txt" file extension which are all the journal entry files using the GetFiles() method of the di object.
            Console.WriteLine(""); // creates some line spacing.
            Console.WriteLine("Here are the saved journal entries: "); // shows the user the listing of all the journal entry files that were saved.
            Console.WriteLine(""); // creates some line spacing.
            foreach (FileInfo file in files) { // the code in this for each loop will run as long there is a file to be listed in the FileInfo object called file with each file from files.
                Console.WriteLine(file.Name); // shows the user what the list of the names of what the saved journal entry files are using file.Name.
            } // the end of the for each loop.
            string path = "JournalSystem.exe"; // creates a string variable called path which is initialized to the name of the program's exe file name.
            FileInfo f = new FileInfo(path); // creates a new FileInfo object called f with path as it's argument.
            Console.WriteLine(""); // creates some line spacing.
            Console.WriteLine("And the main system is saved here: "); // shows the user where the main exe file of the program is saved.
            Console.WriteLine(""); // creates some line spacing.
            Console.WriteLine(f.FullName); // shows the user the path to where the main exe file of the program is saved using f.fullName.
            Console.WriteLine(""); // creates some line spacing.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.WriteLine(""); // creates some line spacing.
            Console.WriteLine("Listing completed. Returning to Main Menu..."); // tells the user that the listing has been completed and the program is returning to the main menu.
            path = ""; // the value of path gets reset to an empty string.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.Clear(); // clears the window by calling the Console.Clear() method.
        } // the end of the listJournalEntry() method.

        /*
        * this public static void deleteJournalEntry() file deletes a journal entry file that the user specifies.
        * this method is void because it doesn't return anything.
        */

        public static void deleteJournalEntry() {
            Console.Write("Enter the name of the journal entry: "); // prompts the user to enter the name of the journal entry file that they want to have deleted.
            fileName = Console.ReadLine(); // fileName gets the user input saved as a string using the Console.ReadLine() method.
            Console.WriteLine(""); // creates some line spacing.
            FileInfo fileInfo = new FileInfo(fileName); // creates a new FileInfo object called fileInfo with fileName as it's argument.
            File.Delete(fileInfo.FullName); // deletes the file specified inside the File.Delete() method argument fileInfo.FullName which has the file path from fileInfo.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.WriteLine("Journal Entry deleted. Returning to Main Menu..."); // tells the user that the file has been deleted and that the program is returning to the main menu.
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.Clear(); // clears the window by calling the Console.Clear() method.
        } // the end of the deleteJournalEntry() file.

        /*
        * this static public void Main() method runs the code for the program.
        * this method is void because it doesn't return anything.
        */

        static public void Main(string[] args) {
            Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
            Console.Clear(); // clears the window by calling the Console.Clear() method.
            Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
            Console.WriteLine("----- Welcome to the Journal System -----"); // shows the heading of the main menu of the program.
            Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
            Console.WriteLine("Please choose from one of the following: "); // prompts the user to choose one of the following options from the list of options for the program.
            Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
            Console.WriteLine("1: Create and Write Journal Entry"); // shows the user that option 1 creates and writes journal entries.
            Console.WriteLine("2: Read Journal Entry"); // shows the user that option 2 reads journal entries.
            Console.WriteLine("3: Exit System"); // shows the user that option 3 exits the system.
            Console.WriteLine("4: List Journal Entries and Folder"); // shows the user that option 4 lists the journal entries and where the program file is saved at.
            Console.WriteLine("5: Delete Journal Entry"); // shows the user that option 5 deletes journal entries.
            Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
            Console.Write("Choice: "); // prompts the user to enter a number that corresponds to one of the 5 choices which are the 5 functions of the program.
            choice = int.Parse(Console.ReadLine()); // choice gets the user input saved as a integer number using the Console.ReadLine() method.
            while (choice != 3) { // the code inside the while loop runs as long as the value inside choice isn't a 3.
                if (choice == 1) { // the code inside the if statement gets run if the value of choice was a 1.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    createJournalEntry(); // calls the createJournalEntry() method.
                } // the end of the if statement.
                else if (choice == 2) { // the code inside the else if statement gets run if the value of choice was a 2.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    readJournalEntry(); // calls the readJournalEntry() method.
                } // the end of the else if statement.
                else if (choice == 3) { // the code inside the else if statement gets run if the value of choice was a 3.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    exitSystem(); // calls the exitSystem() method.
                } // the end of the else if statement.
                else if (choice == 4) { // the code inside the else if statement gets run if the value of choice was a 4.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    listJournalEntry(); // calls the listJournalEntry() method.
                } // the end of the else if statement.
                else if (choice == 5) { // the code inside the else if statement gets run if the value of choice was a 5.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    deleteJournalEntry(); // calls the deleteJournalEntry() method.
                } // the end of the else if statement.
                Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
                Console.Write("Do you want to use the Journal System again (y/n)? "); // prompts the user ot enter either a y or n based on whether they want to use the program again or not.
                exitCondition = Console.ReadLine(); // exitCondition gets the user input saved as a string using the Console.ReadLine() method.
                Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                Console.Clear(); // clears the window by calling the Console.Clear() method.
                if (exitCondition == "y") { // the code inside the if statement if the value of exitCondition was a y, which means the user wants to use the program again.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    Console.WriteLine("----- Welcome to the Journal System -----"); // shows the heading of the main menu of the program.
                    Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
                    Console.WriteLine("Please choose from one of the following: "); // prompts the user to choose one of the following options from the list of options for the program.
                    Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
                    Console.WriteLine("1: Create and Write Journal Entry"); // shows the user that option 1 creates and writes journal entries.
                    Console.WriteLine("2: Read Journal Entry"); // shows the user that option 2 reads journal entries.
                    Console.WriteLine("3: Exit System"); // shows the user that option 3 exits the system.
                    Console.WriteLine("4: List Journal Entries and Folder"); // shows the user that option 4 lists the journal entries and where the program file is saved at.
                    Console.WriteLine("5: Delete Journal Entry"); // shows the user that option 5 deletes journal entries.
                    Console.WriteLine(""); // clears the window by calling the Console.Clear() method.
                    Console.Write("Choice: "); // prompts the user to enter a number that corresponds to one of the 5 choices which are the 5 functions of the program.
                    choice = int.Parse(Console.ReadLine()); // choice gets the user input saved as a integer number using the Console.ReadLine() method.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                } // the end of the if statement.
                else if (exitCondition == "n") { // the code inside the else if statement gets run if the value of exitCondition was a n, which means the yser doesn't want to use the program again.
                    Thread.Sleep(2000); // slows down the speed at which the program runs by 2000 milliseconds (2 seconds) so that it looks nice and neat and not too fast and so that the user can actually see and understand what is going on.
                    resetValues(); // calls the resetValues() method.
                    Console.Clear(); // clears the window by calling the Console.Clear() method.
                    exitSystem(); // calls the exitSystem() method.
                } // the end of the else if statement.
            } // the end of the while loop.
        } // the end of the Main() method.
    } // the end of the JournalSystem class.
} // the end of the Journal namespace.